dataconnectors/influxdb/influxdb.go

```diff
@@ -124,12 +136,28 @@ func (c *InfluxDbConnector) refreshData(epoch time.Time, period time.Duration, i
 	c.dataMutex.Lock()
 	defer c.dataMutex.Unlock()
 
-	periodStart := epoch
-	periodEnd := epoch.Add(period)
-
-	if !c.lastFetchPeriodEnd.IsZero() && c.lastFetchPeriodEnd.After(epoch) {
-		// If we've already fetched, only fetch the difference with an interval overlap
-		periodStart = c.lastFetchPeriodEnd.Add(-interval)
+	var periodStart time.Time
+	var periodEnd time.Time
+
+	if epoch.IsZero() {
+		// Epoch not set - sliding window from now
+		if c.lastFetchPeriodEnd.IsZero() {
+			// fetch period from now
+			periodStart = now().UTC().Add(-period)
+			periodEnd = periodStart.Add(period)
+		} else {
+			// If we've already fetched, only fetch the difference with an interval overlap
```

Comment:
After actually firing this up and testing it, I think this logic is wrong - we want to first fetch a period's worth of data looking back from now (above), then fetch a *fetch interval*'s worth of data each time we are invoked and push it.

Actually running this with the serverops params:

```
period: 24h
interval: 10m
granularity: 30s
```

Produces queries at the refresh interval (15s) that are a day wide and a day further into the future each time:

```
		lastFetchPeriodEnd: 2021-10-06T21:59:28Z

		from(bucket:"my-bucket") |>
		range(start: 2021-10-06T21:49:28Z, stop: 2021-10-07T21:59:28Z) |>
		filter(fn: (r) => r["_measurement"] == "cpu") |>
		filter(fn: (r) => r["_field"] == "usage_idle") |>
		aggregateWindow(every: 10m0s, fn: mean, createEmpty: false)
    
		lastFetchPeriodEnd: 2021-10-07T21:59:28Z

		from(bucket:"my-bucket") |>
		range(start: 2021-10-07T21:49:28Z, stop: 2021-10-08T21:59:28Z) |>
		filter(fn: (r) => r["_measurement"] == "cpu") |>
		filter(fn: (r) => r["_field"] == "usage_idle") |>
		aggregateWindow(every: 10m0s, fn: mean, createEmpty: false)
    
		lastFetchPeriodEnd: 2021-10-08T21:59:28Z

		from(bucket:"my-bucket") |>
		range(start: 2021-10-08T21:49:28Z, stop: 2021-10-09T21:59:28Z) |>
		filter(fn: (r) => r["_measurement"] == "cpu") |>
		filter(fn: (r) => r["_field"] == "usage_idle") |>
		aggregateWindow(every: 10m0s, fn: mean, createEmpty: false)
```

Maybe you're blending the concept of "interval" with "fetchInterval"?

I think what we'd actually want is something like:

```
periodStart = c.lastFetchPeriodEnd
periodEnd = now().UTC()
```

Which will continue streaming data in at the refresh interval.

Reply:
Yeah, it was just a miss on period end, which should be now. Will fix.